app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -188,15 +189,29 @@ def setup_agent(instance)
 
       docker_username = docker_auth.userid
       docker_password = docker_auth.password
-      command_line = "docker login"
-      command_line << " #{docker_image}"
-      command_line << " -u #{docker_username} -p #{docker_password}"
-      ssh.perform_commands([command_line])
+      login_params = {
+        nil  => docker_image,
+        '-u' => docker_username,
+        '-p' => docker_password
+      }
+      login_cmd = AwesomeSpawn.build_command_line("sudo docker login", login_params)
+
+      ssh.perform_commands([login_cmd])
```

Comment:
This will cause the password to be on the command line, and thus printed to the log on error.  We should avoid putting passwords on the command line.